lib/tdf3/src/client/builders.ts

```diff
@@ -794,8 +794,25 @@ class DecryptParamsBuilder {
   /**
    * @param rcaParams
    */
-  setRcaSource(rcaParams: RcaParams) {
-    this._params.rcaSource = rcaParams;
+  setRcaSource(rcaParams: RcaParams | RcaLink) {
+    if (typeof rcaParams === 'object') {
+      this._params.rcaSource = rcaParams;
+    } else if (typeof rcaParams === 'string') {
+      try {
+        const rcaLink: URLSearchParams = new URLSearchParams(rcaParams);
+        const pu = rcaLink.get('pu');
+        const wu = rcaLink.get('wu'); // link for download
+        const wk = rcaLink.get('wk'); // split knowledge
+        const al = rcaLink.get('al');
+        if (!pu || !wu || !wk || !al) {
+          throw new Error('RCA link is not valid');
+        }
+        this._params.rcaSource = { pu, wu, wk, al };
+      } catch (e) {
+        console.error('RCA link is not valid', e);
+        throw new Error('RCA link is not valid');
+      }
```

Comment:
Reorganize the logic so you aren't re-throwing and include more details.
```suggestion
        const al = rcaLink.get('al');
      } catch (cause) {
        throw new Error('RCA link is not valid', {cause});
      }
      if (!pu || !wu || !wk || !al) {
        throw new Error(`RCA link [${rcaParams}] is missing parameters!`);
      }
      this._params.rcaSource = { pu, wu, wk, al };
```